dot/network/host.go

```diff
@@ -65,15 +63,27 @@ func newHost(ctx context.Context, cfg *Config) (*host, error) {
 	}
 
 	var externalAddr ma.Multiaddr
-	ip, err := pubip.Get()
-	if err != nil {
-		logger.Errorf("failed to get public IP: %s", err)
-	} else {
-		logger.Debugf("got public IP %s", ip)
+	if cfg.PublicIP != "" {
+		ip := net.ParseIP(cfg.PublicIP)
+		if ip == nil {
+			return nil, fmt.Errorf("invalid public ip: %s", cfg.PublicIP)
+		}
+		logger.Debugf("using config PublicIP: %s", ip)
 		externalAddr, err = ma.NewMultiaddr(fmt.Sprintf("/ip4/%s/tcp/%d", ip, cfg.Port))
 		if err != nil {
 			return nil, err
 		}
+	} else {
+		ip, err := pubip.Get()
+		if err != nil {
+			logger.Errorf("failed to get public IP error: %v", err)
```

Comment:
should we return error here?

If we are here, that means we are failing to set `externalAddr`. `externalAddr` is getting used in the code after this, but not getting set. Looks like we should return the error here.

Reply:
This is how the code was.  In the event we can't get a public IP via `pubip` I would assume nothing gets broadcasted.  What do you think @noot?

Reply:
```
		libp2p.AddrsFactory(func(as []ma.Multiaddr) []ma.Multiaddr {
			addrs := []ma.Multiaddr{}
			for _, addr := range as {
				if !privateIPs.AddrBlocked(addr) {
					addrs = append(addrs, addr)
				}
			}
			if externalAddr == nil {
				return addrs
			}

			return append(addrs, externalAddr)
		}),
```

other addresses can still be advertised even if `pubip.Get` fails, so I think that's why I didn't make it return an error